ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/http/JsonRpcErrorHandler.java

```diff
@@ -12,84 +12,53 @@
+    final JsonRpcRequestId requestId = context.get(JsonRpcHandler.ID_ELEMENT_NAME);
     final int statusCode =
         context.statusCode() == -1 ? INTERNAL_SERVER_ERROR.code() : context.statusCode();
-    LOG.debug(
-        "Failed to correctly handle request. method: {}, uri: {}, body: {}, Error body: {}",
-        context.request()::method,
-        context.request()::absoluteURI,
-        () -> jsonRpcRequest.map(Json::encodePrettily).orElse(context.getBodyAsString()),
-        () -> Json.encode(errorResponse));
-    LOG.error("Exception = ", context.failure());
-    httpResponseFactory.response(context.response(), statusCode, errorResponse);
-  }
-
-  private Optional<JsonRpcRequest> jsonRpcRequest(final RoutingContext context) {
-    try {
-      return Optional.of(jsonDecoder.decodeValue(context.getBody(), JsonRpcRequest.class));
-    } catch (final DecodeException e) {
-      LOG.debug("Parsing body as JSON failed for: {}", context.getBodyAsString(), e);
-      return Optional.empty();
-    }
-  }
-
-  private JsonRpcErrorResponse errorResponse(
-      final RoutingContext context, final Optional<JsonRpcRequest> jsonRpcRequest) {
-    final JsonRpcRequestId rpcRequestId =
-        jsonRpcRequest.map(JsonRpcRequest::getId).orElse(new JsonRpcRequestId(null));
-    final JsonRpcError jsonRpcError = jsonRpcError(context);
-    return new JsonRpcErrorResponse(rpcRequestId, jsonRpcError);
-  }
 
-  private JsonRpcError jsonRpcError(final RoutingContext context) {
-    if (context.failure() instanceof JsonRpcException) {
-      final JsonRpcException jsonRpcException = (JsonRpcException) context.failure();
-      return jsonRpcException.getJsonRpcError();
-    } // in case of a timeout we may not have a failure exception so we use the status code
-    else if (context.statusCode() == BAD_GATEWAY.code()) {
-      return FAILED_TO_CONNECT_TO_DOWNSTREAM_NODE;
-    } else if (context.statusCode() == GATEWAY_TIMEOUT.code()) {
-      return CONNECTION_TO_DOWNSTREAM_NODE_TIMED_OUT;
+    final Throwable failure = context.failure();
+    if (failure instanceof JsonRpcException) {
+      final JsonRpcException ex = (JsonRpcException) context.failure();
+      httpResponseFactory.failureResponse(
+          context.response(), requestId, statusCode, ex.getJsonRpcError());
+    } else if (failure instanceof ConnectException || failure instanceof SSLException) {
+      httpResponseFactory.failureResponse(
+          context.response(),
+          requestId,
+          statusCode,
+          JsonRpcError.FAILED_TO_CONNECT_TO_DOWNSTREAM_NODE);
+    } else if (failure instanceof TimeoutException) {
+      httpResponseFactory.failureResponse(
+          context.response(),
+          requestId,
+          statusCode,
+          JsonRpcError.CONNECTION_TO_DOWNSTREAM_NODE_TIMED_OUT);
     } else {
-      return INTERNAL_ERROR;
+      context.response().setStatusCode(statusCode);
```

Comment:
Think we should be consistent always return a JsonRpcError, so in this case could return a JsonRpcError with INTERNAL_ERROR and a null id

Reply:
sure.